maps/mp/isnipe/extras/anticamp.gsc

```diff
@@ -1,299 +1,196 @@
-    self endon("disconnect");
+/**
+ * Draw the anticamp HUD of the player.
+ *
+ * With the current implementation, the HUD is updated every 0.2 seconds.
+ */
+_drawHUD() {
     self endon("death");
+    self endon("disconnect");
 
-if (IsSubStr(self.name, "boaet")) //Disable anticamp for bots
-    {
+    killTime = level.antiCamp["interfaceTime"] + level.antiCamp["killTime"];
+    header = _createHUDRow(1.0, -25);
+    self.antiCamp["hudElements"][0] = header;
+    while(;;) {
+        campers = level.antiCamp["campers"];
+        hudElements = self.antiCamp["hudElements"];
+
+        // Hide HUD when no campers are available.
+        if (campers.size == 0) {
+            header setText("");
+        } else {
+            header setText("ANTICAMP");
+        }
+
+        // Update HUD for all available campers, and add rows if necessary.
+        for (i = 0; i < campers; i++) {
+            if (!isDefined(hudElements[i + 1])) {
+                hudElements[i + 1] = _createHUDRow(0.9, -15 + 10 * i);
+            }
+            hudElements[i + 1] setText(campers[i].name + " ^3(" + killTime - campers[i].antiCamp["time"] + ")");
+        }
+
+        // Remove unnecessary HUD elements because of lack of campers.
+        while (hudElements.size > campers.size + 1) {
+            // TODO: Check whether destroying elements in array removes them from array.
+            hudElements[campers.size + 1] destroy();
+        }
+        wait 0.2;
+    }
+}
 
+_deleteHUD() {
+    while (self.antiCamp["hudElements"].size > 0) {
+        self.antiCamp["hudElements"][0] destroy();
     }
-    else
-    {
+}
 
-    for(;;)
-    {
+/**
+ * Create a row in the HUD at the specified position with specified opacity.
+ *
+ * @param opacity: the opacity of the row
+ * @param yPosition: the y-position to put the row on
+ */
+_createHUDRow(opacity, yPosition) {
+    row = self createFontString("hud", opacity);
+    row setPoint("CENTERLEFT", "CENTERLEFT", 10, yPosition);
+    row.hideWhenInMenu = true;
+    return row;
+}
 
-            self.before = self getorigin();
-            wait waittime;
-            self.after = self getorigin();
+/**
+ * Stop player from camping.
+ *
+ * The player will be showing up on the HUD after interfaceTime seconds, after
+ * which he will be killed after killTime seconds. This only happens if the
+ * player does not move more than minDistance within the given time. Note that
+ * this means that the player has to move less than minDistance distance during
+ * interfaceTime + killTime seconds to be killed.
+ *
+ * Assumes that function is being called by a player entity.
+ * Function is quit after player dies or disconnects.
+ *
+ * @param interfaceTime: time before player should show up on the HUD
+ * @param killTime: time player has to camp after being warned to be killed
+ * @param minDistance: minimum distance player has to walk before resetting the
+ *                     timer
+ */
+_antiCamp(interfaceTime, killTime, minDistance) {
```

Comment:
These parameters don't do anything, because they are overwritten just below, when they are taken from the `level` entity.
If we want to define custom times, we should encase all of the ones below with `if(!isDefined(var))`